compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/SwingPanel.desktop.kt

```diff
@@ -149,9 +161,76 @@ internal class SwingInteropViewGroup(
         }
         isFocusCycleRoot = true
     }
+
+    override fun doLayout() {
+        components[0].setBounds(0, 0, width, height)
+    }
+
+    override fun getPreferredSize(): Dimension? {
+        return components[0].preferredSize
+    }
+
+    override fun getMinimumSize(): Dimension? {
+        return components[0].minimumSize
+    }
+
+    override fun getMaximumSize(): Dimension? {
+        return components[0].maximumSize
+    }
+
+    override fun invalidate() {
+        super.invalidate()
+        onInvalidate?.invoke()
+    }
 }
 
 /**
  * The maximum radix available for conversion to and from strings.
  */
-private val MaxSupportedRadix = 36
+private const val MaxSupportedRadix = 36
+
+private class AwtContentMeasurePolicy(
+    val component: Component,
+) : MeasurePolicy {
+
+    private fun Density.awtToPx(awtPx: Int): Int = (awtPx * density).roundToInt()
+
+    override fun MeasureScope.measure(
+        measurables: List<Measurable>,
+        constraints: Constraints
+    ): MeasureResult {
+        val prefSize = component.preferredSize
+        val width = awtToPx(prefSize.width).coerceIn(constraints.minWidth, constraints.maxWidth)
+        val height = awtToPx(prefSize.height).coerceIn(constraints.minHeight, constraints.maxHeight)
+        return layout(width, height) {}
+    }
+
+    override fun IntrinsicMeasureScope.minIntrinsicWidth(
+        measurables: List<IntrinsicMeasurable>,
+        height: Int
+    ): Int {
+        return awtToPx(component.minimumSize.width)
```

Comment:
Required helper function is alredy exist
```suggestion
        return component.minimumSize.width.dp.roundToPx()
```

Reply:
Changed the implementation of `awtToPx`. I don't want to repeat `.dp.roundToPx` in 6 places.